apps/server/src/modules/user/uc/user.uc.ts

```diff
@@ -1,37 +1,32 @@
-import { BadRequestException, Injectable } from '@nestjs/common';
-import { ConfigService } from '@nestjs/config';
-import { EntityId, LanguageType, PermissionService, User } from '@shared/domain';
-import { UserRepo } from '@shared/repo';
+import { Injectable } from '@nestjs/common';
+import { EntityId, User } from '@shared/domain';
+import { UserDto } from '@src/modules/user/uc/dto/user.dto';
+import { UserService } from '@src/modules/user/service/user.service';
+import { ProvisioningUserOutputDto } from '@src/modules/provisioning/dto/provisioning-user-output.dto';
+import { UserUcMapper } from '@src/modules/user/mapper/user.uc.mapper';
+import { RoleUc } from '@src/modules/role/uc/role.uc';
+import { RoleDto } from '@src/modules/role/service/dto/role.dto';
 import { ChangeLanguageParams } from '../controller/dto';
-import { IUserConfig } from '../interfaces';
 
 @Injectable()
-export class UserUC {
-	constructor(
-		private readonly userRepo: UserRepo,
-		private readonly permissionService: PermissionService,
-		private readonly configService: ConfigService<IUserConfig, true>
-	) {}
+export class UserUc {
+	constructor(private readonly userService: UserService, private readonly roleUc: RoleUc) {}
 
 	async me(userId: EntityId): Promise<[User, string[]]> {
```

Comment:
We moved the repo call to the userservice. There shouldnt be database repo calls in usecases (other things like a repository interface would be ok). They give you entites back in usecases. In my opinion you should only work on business entites/ domain objects in usecases.

Reply:
I fully and completely agree with that. This has been identified as the nr1 weakness of our current architecture, and we want to work on solving this by introducing a repository interface with domain objects to all repositories once our current copy initiative is completed.

It could be argued that once that refactoring is done, we would want to revert this change to get the orchestration back into the usecase, since then it would be valid.  With that in mind, I think it is acceptable to keep the repo calls in the ucs temporarily, to at least keep the existing code consistent until the refactoring.

An alternative could be to look for ways to already introduce such an interface to the things you need to touch, but I expect that to be a lot of effort...

Reply:
We have rolled back the changes. Now the repositories will be used again